aws/resource_aws_ecs_task_set_test.go

```diff
@@ -0,0 +1,1076 @@
+    type = "EXTERNAL"
+  }
+}
+resource "aws_ecs_task_set" "main" {
+  service 		    = "${aws_ecs_service.main.id}"
+  cluster         = "${aws_ecs_cluster.main.id}"
+  task_definition = "${aws_ecs_task_definition.mongo.arn}"
+  launch_type      = "FARGATE"
+  platform_version = %q
+  network_configuration {
+    security_groups  = ["${aws_security_group.allow_all_a.id}", "${aws_security_group.allow_all_b.id}"]
+    subnets          = ["${aws_subnet.main.*.id[0]}", "${aws_subnet.main.*.id[1]}"]
+    assign_public_ip = false
+  }
+}
+`, sg1Name, sg2Name, clusterName, tdName, svcName, platformVersion)
+}
+
+func testAccCheckAWSEcsTaskSetExists(name string, taskSet *ecs.TaskSet) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ecsconn
+
+		input := &ecs.DescribeTaskSetsInput{
+			TaskSets: []*string{aws.String(rs.Primary.ID)},
+			Cluster:  aws.String(rs.Primary.Attributes["cluster"]),
+			Service:  aws.String(rs.Primary.Attributes["service"]),
+		}
+		var output *ecs.DescribeTaskSetsOutput
+		err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+			var err error
+			output, err = conn.DescribeTaskSets(input)
+
+			if err != nil {
+				if isAWSErr(err, ecs.ErrCodeClusterNotFoundException, "") {
+					return resource.RetryableError(err)
+				}
+				if isAWSErr(err, ecs.ErrCodeServiceNotFoundException, "") {
+					return resource.RetryableError(err)
+				}
+				if isAWSErr(err, ecs.ErrCodeTaskSetNotFoundException, "") {
+					return resource.RetryableError(err)
+				}
+				return resource.NonRetryableError(err)
+			}
+
+			if len(output.TaskSets) == 0 {
+				return resource.RetryableError(fmt.Errorf("task set not found: %s", rs.Primary.ID))
+			}
+
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		*taskSet = *output.TaskSets[0]
+
+		return nil
+	}
+}
+
+func testAccCheckAWSEcsTaskSetDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ecsconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ecs_task_set" {
+			continue
+		}
+
+		out, err := conn.DescribeTaskSets(&ecs.DescribeTaskSetsInput{
+			TaskSets: []*string{aws.String(rs.Primary.ID)},
+			Cluster:  aws.String(rs.Primary.Attributes["cluster"]),
+			Service:  aws.String(rs.Primary.Attributes["service"]),
+		})
+
+		if err == nil {
+			if len(out.TaskSets) > 0 {
+				var activeTaskSets []*ecs.TaskSet
+				for _, ts := range out.TaskSets {
+					if *ts.Status != "INACTIVE" {
+						activeTaskSets = append(activeTaskSets, ts)
+					}
+				}
+				if len(activeTaskSets) == 0 {
+					return nil
+				}
+
+				return fmt.Errorf("ECS task set still exists:\n%#v", activeTaskSets)
+			}
+			return nil
+		}
+
+		return err
+	}
+
+	return nil
+}
+
+func testAccCheckAWSEcsTaskSetDisappears(taskSet *ecs.TaskSet) resource.TestCheckFunc {
```

Comment:
please see https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#disappears-acceptance-tests

and use the the `testAccCheckResourceDisappears` function